### PR TITLE
Amend Nagios/Ansible Config Role To Create Host Groups Config File 

### LINKED ISSUE #2864: Nagios : Amend Nagios/Ansible Config Role To Create Host Groups Config File
Part of epic : [#2445](https://github.com/adoptium/infrastructure/issues/2445) ( Task 6.2 )

Update the Nagios\_Config playbook to create the hostgroups.cfg file

The script should parse the ansible inventory ( similar to issue #2629 )

And create a config file which contains a hostgroup entry for each of the providers in the inventory, ie the second stanza for the build, test & dockerhost level 1 stanzas.  
  
define hostgroup{  
       hostgroup\_name  Ibmcloud  
       alias           Ibmcloud  
       members         ,test-ibmcloud-rhel7-x64-1,test-ibmcloud-rhel6-x64-1,test-ibmcloud-ubuntu1604-x64-1,build-ibmcloud-win2012r2-x64-1,build-ibmcloud-win2012r2-x64-2,test-ibmcloud-win2012r2-x64-1,test-ibmcloud-win2012r2-x64-2  
       }

The playbook should then add the relevant machines to the members list as shown in the example above.

### PR DESCRIPTION
**Pull request by lumuchris256**
Updated the Nagios_Config playbook to create the Hostgroups.cfg file that contains a service group entry for each of the group types in the inventory

Fixes #2864 
signed off by @lumuchris256 
cc @steelhead31 @Haroon-Khel @smlambert

### CODE REVIEW

**Comment by steelhead31** [comment]
This PR will require changes, on a clean run of the playbooks..I get this error..  
  
TASK \[Nagios\_Config : Run Create Nagios Hostgroups script\] \*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*  
fatal: \[localhost\]: FAILED! => {"changed": true, "msg": "non-zero return code", "rc": 1, "stderr": "Traceback (most recent call last):\\n  File \\"/home/vagrant/.ansible/tmp/ansible-tmp-1675167368.6075487-39101-189259814438941/Create\_Nagios\_Hostgroups.py\\", line 20, in \<module>\\n    Nagios\_Service\_Types = sys.argv\[3\]\\nIndexError: list index out of range\\n", "stderr\_lines": \["Traceback (most recent call last):", "  File \\"/home/vagrant/.ansible/tmp/ansible-tmp-1675167368.6075487-39101-189259814438941/Create\_Nagios\_Hostgroups.py\\", line 20, in \<module>", "    Nagios\_Service\_Types = sys.argv\[3\]", "IndexError: list index out of range"\], "stdout": "", "stdout\_lines": \[\]}  
 

I'll do some investigation and try and find out what isnt working.

File: ansible/playbooks/nagios/roles/Nagios_Config/tasks/Create_Nagios_Hostgroups.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Run Create Nagios Hostgroups script
+  script: ../scripts/Create_Nagios_Hostgroups.py "{{Nagios_Base_Path}}/etc/objects" "{{Nagios_Service_Types}}"
```
Comment (steelhead31) [inline ansible/playbooks/nagios/roles/Nagios_Config/tasks/Create_Nagios_Hostgroups.yml:3]:
```suggestion
  script: ../scripts/Create_Nagios_Hostgroups.py "{{Input_Path}}" "{{Nagios_Base_Path}}/etc/objects" "{{Nagios_Service_Types}}"
```

Comment (steelhead31) [reply ansible/playbooks/nagios/roles/Nagios_Config/tasks/Create_Nagios_Hostgroups.yml:3]: Requires the path to the inventory to be added.

File: ansible/playbooks/nagios/roles/Nagios_Config/scripts/Create_Nagios_Hostgroups.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                            if host_type==service:
+                                formatted_name = host_type+'-'+provider_name+'-'+host
+                                if formatted_name not in hostList:
+                                    hostList.append(formatted_name)
+                                    providerList.append(provider_name)
+                    for provider_name in providerList:
+                         if provider_name not in uniqProvider:
+                             uniqProvider.append(provider_name)
+
+    with open('hostgroups.cfg', 'w') as f:
```
Comment (steelhead31) [inline ansible/playbooks/nagios/roles/Nagios_Config/scripts/Create_Nagios_Hostgroups.py:96]:
```suggestion
    with open(f"{Output_Path}/hostgroups.cfg", "w") as f:
```

Comment (steelhead31) [reply ansible/playbooks/nagios/roles/Nagios_Config/scripts/Create_Nagios_Hostgroups.py:96]: Include the relative output pathing.

**Review by steelhead31 — CHANGES_REQUESTED** [review]

**Review by sxa — APPROVED** [review]
Looks reasonable overall, and I'll trust @steelhead31 to check that it works so I'm giving the second approval!

**Pull request merged**